### PR TITLE
Dbs 2.17.3 release

### PR DESCRIPTION
**Pull request by wayne500**

### CODE REVIEW

File: src/main/java/com/backbase/billpay/fiserv/enrolment/EnrolmentMapper.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * Map the phone number to the phone numbers list.
+     * @param dayPhone daytime phone number.
+     * @return list of PhoneNumbers.
+     */
+    default List<PhoneNumber> mapPhoneNumber(String dayPhone) {
+        
+        List<PhoneNumber> phoneNumbers = new ArrayList<PhoneNumber>();
+        phoneNumbers.add(new PhoneNumber().withNumber(dayPhone));
+        return phoneNumbers;
```
Comment (joshuad-bb) [inline src/main/java/com/backbase/billpay/fiserv/enrolment/EnrolmentMapper.java]: What about returning a singletonList here instead?

`    default List<PhoneNumber> mapPhoneNumber(String dayPhone) {
        return Collections.singletonList(dayPhone);
    }`

Comment (wayne500) [reply src/main/java/com/backbase/billpay/fiserv/enrolment/EnrolmentMapper.java]: This suggestion returns a List<String> not a List<PhoneNumber>

File: src/main/java/com/backbase/billpay/fiserv/enrolment/EnrolmentServiceImpl.java [hunk 2]
```diff
@@ -101,4 +107,24 @@ private SubscriberEnrollRequest createEnrolmentRequest(Header header) {
                                                           .build())
             .build();
     }
+
+    @Override
+    public UserByIdGetResponseBody getSubscriberInfo(Header header, String userId) {
+        SubscriberInfoRequest request = createSubscriberInfoRequest(header);
+        LOGGER.debug("Request: {}", request);
+        SubscriberInfoResponse response = client.call(request, SOAP_ACTION_INFO);
+        LOGGER.debug("Received response: {}", response);
```
Comment (sianw) [inline src/main/java/com/backbase/billpay/fiserv/enrolment/EnrolmentServiceImpl.java]: Please remove the response object from the log, perhaps just
`LOGGER.debug("Received response for user: {}, userId);`

File: src/test/java/com/backbase/billpay/fiserv/enrolment/EnrolmentControllerTest.java [hunk 3]
```diff
@@ -54,5 +72,53 @@ public void postBillPayEnrolment() throws Exception {
...
+                        .with(setRemoteAddress())
+                        .param("subscriberID", SUBSCRIBER_ID))
+                        .andExpect(status().isOk())
+                        .andReturn().getResponse().getContentAsString();
+        
+        UserByIdGetResponseBody response =
+                        objectMapper.readValue(stringResponse, UserByIdGetResponseBody.class);
+
+        // validate the subscriber returned matches
+        assertEquals(FIRST, response.getSubscriber().getFirstName());
```
Comment (sianw) [inline src/test/java/com/backbase/billpay/fiserv/enrolment/EnrolmentControllerTest.java]: Can all of the fields be asserted to ensure the mapping is tested?

File: src/main/java/com/backbase/billpay/fiserv/enrolment/EnrolmentMapper.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    
+    /**
+     * Map the phone number to the phone numbers list.
+     * @param dayPhone daytime phone number.
+     * @return list of PhoneNumbers.
+     */
+    default List<PhoneNumber> mapPhoneNumber(String dayPhone) {
+        
+        List<PhoneNumber> phoneNumbers = new ArrayList<PhoneNumber>();
+        phoneNumbers.add(new PhoneNumber().withNumber(dayPhone));
```
[the remark this answers is not in the document]
Comment (wayne500) [reply src/main/java/com/backbase/billpay/fiserv/enrolment/EnrolmentMapper.java]: I've added this

File: src/main/java/com/backbase/billpay/fiserv/enrolment/EnrolmentMapper.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        @Mapping(target = "address.address2", source = "subscriberInformation.usAddress.address2"),
+        @Mapping(target = "address.city", source = "subscriberInformation.usAddress.city"),
+        @Mapping(target = "address.state", source = "subscriberInformation.usAddress.state"),
+        @Mapping(target = "address.postalCode", source = "subscriberInformation.usAddress.zip5"),
+        @Mapping(target = "address.country", ignore = true),
+        @Mapping(target = "address.additions", ignore = true),
+        @Mapping(target = "phoneNumbers", source = "subscriberInformation.dayPhone"),
+        @Mapping(target = "accounts", ignore = true),
+        @Mapping(target = "business", ignore = true),
+        @Mapping(target = "language", ignore = true),
```
[the remark this answers is not in the document]
Comment (wayne500) [reply src/main/java/com/backbase/billpay/fiserv/enrolment/EnrolmentMapper.java]: I've added this.

File: src/main/java/com/backbase/billpay/fiserv/enrolment/EnrolmentServiceImpl.java (hunk 2 above)
Comment (wayne500) [reply src/main/java/com/backbase/billpay/fiserv/enrolment/EnrolmentServiceImpl.java]: Done

File: src/test/java/com/backbase/billpay/fiserv/enrolment/EnrolmentControllerTest.java (hunk 3 above)
Comment (wayne500) [reply src/test/java/com/backbase/billpay/fiserv/enrolment/EnrolmentControllerTest.java]: They sure can. Done

**Pull request merged**